src/convertToObject.js

```diff
@@ -6,7 +6,29 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  let lines = sourceString.split(';');
+  let temp;
+  const result = {};
+
+  lines = lines.filter((string) => string.trim() !== '');
+
+  if (!sourceString) {
+    return {};
+  }
+
+  if (lines.length === 1) {
+    temp = lines[0].split(':');
+    result[temp[0].trim()] = temp[1].trim();
+
+    return result;
+  }
+
+  for (let i = 0; i < lines.length; i++) {
+    temp = lines[i].split(':');
+    result[temp[0].trim()] = temp[1].trim();
+  }
```

Comment:
This logic is duplicates now, you need to simplify it and use the forEach method instead if for